input/tangy-select.js

```diff
@@ -143,11 +143,52 @@ class TangySelect extends TangyInputBase {
     }
   }
 
+  // this method generate object id for xAPI statement based on form, item, and input ids
+  _generateObjectId() {
+    const formEl = this.closest('tangy-form');
+    const itemEl = this.closest('tangy-form-item');
+    const formId = formEl ? formEl.id : 'unknown-form';
+    const itemId = itemEl ? itemEl.id : 'unknown-item';
+    const inputId = this.name || this.id;
+    const baseUrl = window.location.origin + window.location.pathname.replace(/\/$/, '');
+    return `${baseUrl}/${formId}/${itemId}/${inputId}`;
+  };
+
   connectedCallback() {
     super.connectedCallback()
     const observer = new MutationObserver(this.render.bind(this))
     observer.observe(this, { attributes: true, childList: true, subtree: true })
     this.render()
+    const options = Array.from(this.shadowRoot.querySelectorAll('select option'))
+    .filter(opt => opt.value);
+
+    const choices = options.map(option => ({
+      id: option.value,
+      description: { 'en-US': option.textContent.trim() }
+    }));
+
+    // clean the label of any HTML tags
+    const label = (this.label || this.name).replace(/<[^>]*>?/gm, '');
```

Comment:
This regex is not clear or explainable: it is therefor not suitable. Should use DOM text functions or similar to do this.